packages/components/inputs/select-utils/src/types.ts

```diff
@@ -0,0 +1,15 @@
+import type SelectComponents from 'react-select';
+
+type ComponentProps<T> = T extends React.ComponentType<infer P> ? P : never;
+
+export type TReactSelectCustomComponentsProps = {
```

Comment:
This is a mapped type that transforms each property in the SelectComponents type to its corresponding prop type. It does this by iterating over each key K in SelectComponents and mapping it to ComponentProps<SelectComponents[K]>. This results in an object type where each property is the prop type of the corresponding component in SelectComponents. The [keyof SelectComponents] part then creates a union of all these prop types. If a new key is added to SelectComponents by the react-select team, TReactSelectCustomComponentsProps will automatically include the prop type for the new key.